src/pages/[platform]/build-a-backend/auth/advanced-workflows/index.mdx

```diff
@@ -0,0 +1,789 @@
+
+```jsx
+import React, { useEffect } from 'react';
+import jwt from 'jwt-decode';
+import {
+  fetchAuthSession,
+} from 'aws-amplify/auth';
+
+const SignInWithGoogle = () => {
+  useEffect(() => {
+  // Check for an existing Google client initialization
+    if (!window.google?.accounts) createScript();
+  }, []);
+
+  // Load the Google client
+  const createScript = () => {
+    const script = document.createElement('script');
+    script.src = 'https://accounts.google.com/gsi/client';
+    script.async = true;
+    script.defer = true;
+    script.onload = initGsi;
+    document.body.appendChild(script);
+  }
+
+  // Initialize Google client and render Google button
+  const initGsi = () => {
+    if (window.google?.accounts) {
+      window.google.accounts.id.initialize({
+        client_id: process.env.GOOGLE_CLIENT_ID,
+        callback: (response: any) => {
+          customCredentialsProvider.loadFederatedLogin({
+            domain: 'accounts.google.com',
+            token: response.credential,
+          });
+          const fetchSessionResult = await fetchAuthSession(); // will return the credentials
+          console.log('fetchSessionResult: ', fetchSessionResult);
+        },
+      });
+      window.google.accounts.id.renderButton(
+        document.getElementById('googleSignInButton'),
+        { theme: 'outline', size: 'large' }
+      );
+    }
+  }
+
+  return (
+    <div>
+      <button id='googleSignInButton'/>
+    </div>
+  );
+}
+```
+
+### Federate with Auth0
+
+You can use `Auth0` as one of the providers of your Cognito Identity Pool. This will allow users authenticated via Auth0 have access to your AWS resources.
+
+Step 1. [Follow Auth0 integration instructions for Cognito Federated Identity Pools](https://auth0.com/docs/integrations/integrating-auth0-amazon-cognito-mobile-apps)
+
+Step 2. Login with `Auth0`, then use the id token returned to get AWS credentials from `Cognito Federated Identity Pools` using custom credentials provider you created at the start:
+
+```js
+import { fetchAuthSession } from 'aws-amplify/auth';
+
+const { idToken, domain, name, email, phoneNumber } = getFromAuth0(); // get the user credentials and info from auth0
+
+async function getCognitoCredentials() {
+  try {
+    customCredentialsProvider.loadFederatedLogin({
+      domain,
+      token: idToken
+    });
+    const fetchSessionResult = await fetchAuthSession(); // will return the credentials
+    console.log('fetchSessionResult: ', fetchSessionResult);
+  } catch (err) {
+    console.log(err);
+  }
+}
+```
+
+</InlineFilter>
+
+## Lambda Triggers
+
+The CLI allows you to configure [Lambda Triggers](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html) for your Cognito User Pool. These enable you to add custom functionality to your registration and authentication flows. [Read more](/[platform]/build-a-backend/functions/set-up-function/)
```

Comment:
change this to "in `defineAuth` and then link to our docs page with the triggers